### PR TITLE
Add assemblies, test for checking empty spawn points and refactor PlayerStats

### PR DESCRIPTION
**Pull request by Cavitedev**
I still need more refactoring to apply Single responsibility principle and other improvements on the refactor code

### CODE REVIEW

File: Shootingat Resolved Project/Assets/Tests/EditorTests/ValidateRooms.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            string[] filesPaths = Directory.GetFiles(roomPath);
+            List<String> prefabFilesPaths = filesPaths.Where(file => file.EndsWith("prefab")).ToList();
+
+            foreach (string prefabPath in prefabFilesPaths)
+            {
+                GameObject prefab = AssetDatabase.LoadAssetAtPath<GameObject>(prefabPath);
+                Room room = prefab.GetComponent<Room>();
+
+
+                var bindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance;
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Tests/EditorTests/ValidateRooms.cs:44]: Could not extract method for reflection. This is an appalling practice for testing.

File: Shootingat Resolved Project/Assets/Tests/EditorTests/ValidateRooms.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+using System.Linq;
+using System.Reflection;
+using NUnit.Framework;
+using PabloLario.DungeonGeneration;
+using UnityEditor;
+using UnityEngine;
+
+
+
+public class ValidateRooms
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Tests/EditorTests/ValidateRooms.cs:14]: This is not unit testing. This is verification of assets so it does not test any actual code

File: Shootingat Resolved Project/Assets/Scripts/Characters/Player/Powerups/HealthPowerup.cs [hunk 3]
```diff
@@ -8,12 +8,12 @@ public class HealthPowerup : Powerup
     {
         [SerializeField] private int addedHealth;
 
-        public override void ApplyPowerup()
+        public override void ApplyPowerup(PlayerStats ps)
         {
             if (goodPowerup)
-                PlayerStats.Instance.ModifyClarity(addedHealth);
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Scripts/Characters/Player/Powerups/HealthPowerup.cs:14]: Split method to be more clear about whether it is improving it or making it worse

File: Shootingat Resolved Project/Assets/Scripts/Characters/Player/PlayerController.cs [hunk 4]
```diff
@@ -1,13 +1,13 @@
 using UnityEngine;
 using UnityEngine.SceneManagement;
 using PabloLario.Powerups;
-using PabloLario.Shooting;
+using PabloLario.Characters.Core.Shooting;
 using PabloLario.Managers;
 // using Random = UnityEngine.Random; // Un-comment in case I add using System
 
-#pragma warning disable CS0414 // Quitar miembros privados no le�dos
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Scripts/Characters/Player/PlayerController.cs:7]: This is a change that is coming automatically. Maybe it is because I am using Visual Studio code

File: Shootingat Resolved Project/Assets/Scripts/Characters/Core/Stats/Stat.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
+using System;
+using TMPro;
+using UnityEngine;
+
+namespace PabloLario.Characters.Core.Stats
+{
+
+    // Forcing generic type to be number is not available until C# 10
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Scripts/Characters/Core/Stats/Stat.cs]: I would loved to avoid `IntStat` and `FloatStat` classes using a generic parameter which could apply numerical operations

File: Shootingat Resolved Project/Assets/Scripts/Characters/Core/Stats/Stat.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        private TMP_Text textUi;
+
+        [SerializeField]
+        private string textFixture;
+
+        public delegate void OnUpdate(Stat<T> previous, Stat<T> after);
+        public OnUpdate onUpdateValue;
+
+
+        protected void UpdateText()
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Scripts/Characters/Core/Stats/Stat.cs]: Need to refactor this. It is responsible of both UI and stats right now. That's not good

File: Shootingat Resolved Project/Assets/Scripts/Characters/Core/Shooting/Bullet.cs [hunk 7]
```diff
@@ -14,12 +14,10 @@ public class Bullet : MonoBehaviour
         private Vector2 dir = Vector2.zero; // Direction to point to
         private Vector2 initialPos = Vector2.zero;
 
-        private PlayerStats ps;
+        public int damage {get; set;}
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Scripts/Characters/Core/Shooting/Bullet.cs]: Need to refactor this in a BulletStats class

File: Shootingat Resolved Project/Assets/Scripts/Characters/Enemies/EnemyShootBasic.cs [hunk 8]
```diff
@@ -16,6 +16,9 @@ public class EnemyShootBasic : EnemyFollow
         [SerializeField] private float minDistanceAwayFromPlayer;
         [SerializeField] private float shootDst;
 
+        [SerializeField] private int bulletDamage;
```
Comment (Cavitedev) [inline Shootingat Resolved Project/Assets/Scripts/Characters/Enemies/EnemyShootBasic.cs]: Needs to be refactored. This was just to fix the bug of shooting with player stats.

**Comment by Cavitedev** [comment]
Now it is ready to be merged with no conflicts

**Review by Thelario — APPROVED** [review]

**Comment by Thelario** [comment]
Merging pull request from Carlos
